src/System.ComponentModel.TypeConverter/src/System/Drawing/ColorConverter.cs

```diff
@@ -0,0 +1,353 @@
+        {
+            string strValue = value as string;
+            if (strValue != null)
+            {
+                object obj = null;
+                string text = strValue.Trim();
+
+                if (text.Length == 0)
+                {
+                    obj = Color.Empty;
+                }
+                else
+                {
+                    // First, check to see if this is a standard name.
+                    //
+                    obj = ColorTable.GetNamedColor(text);
+
+                    if (obj == null)
+                    {
+                        if (culture == null)
+                        {
+                            culture = CultureInfo.CurrentCulture;
+                        }
+
+                        char sep = culture.TextInfo.ListSeparator[0];
+                        bool tryMappingToKnownColor = true;
+
+                        TypeConverter intConverter = TypeDescriptor.GetConverter(typeof(int));
+
+                        // If the value is a 6 digit hex number only, then
+                        // we want to treat the Alpha as 255, not 0
+                        //
+                        if (text.IndexOf(sep) == -1)
+                        {
+
+                            // text can be '' (empty quoted string)
+                            if (text.Length >= 2 && (text[0] == '\'' || text[0] == '"') && text[0] == text[text.Length - 1])
+                            {
+                                // In quotes means a named value
+                                string colorName = text.Substring(1, text.Length - 2);
+                                obj = Color.FromName(colorName);
+                                tryMappingToKnownColor = false;
+                            }
+                            else if ((text.Length == 7 && text[0] == '#') ||
+                                     (text.Length == 8 && (text.StartsWith("0x") || text.StartsWith("0X"))) ||
+                                     (text.Length == 8 && (text.StartsWith("&h") || text.StartsWith("&H"))))
+                            {
+                                // Note: ConvertFromString will raise exception if value cannot be converted.
+                                obj = Color.FromArgb(unchecked((int)(0xFF000000 | (uint)(int)intConverter.ConvertFromString(context, culture, text))));
+                            }
+                        }
+
+                        // Nope.  Parse the RGBA from the text.
+                        //
+                        if (obj == null)
+                        {
+                            string[] tokens = text.Split(new char[] { sep });
+                            int[] values = new int[tokens.Length];
+                            for (int i = 0; i < values.Length; i++)
+                            {
+                                values[i] = unchecked((int)intConverter.ConvertFromString(context, culture, tokens[i]));
```

Comment:
Seems like this will throw if the conversion can't be performed... is that expected?